StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift

```diff
@@ -411,9 +411,16 @@ extension EmbeddedPaymentElement {
         guard !hasConfirmedIntent else {
             return (.failed(error: PaymentSheetError.embeddedPaymentElementAlreadyConfirmedIntent), nil)
         }
-        // Wait for the last update to finish and fail if didn't succeed. A failure means the view is out of sync with the intent and could e.g. not be showing a required mandate.
-        if let latestUpdateTask {
-            switch await latestUpdateTask.value {
+
+        if let latestUpdateContext {
+            switch latestUpdateContext.status {
+            case .inProgress:
+                // Fail confirmation immediately if an update is in progress, rather than waiting for it to complete.
+                // This prevents scenarios where the user might confirm an outdated state, such as agreeing to pay X
+                // but actually being charged Y due to an in-flight update changing the amount.
+                let errorMessage = "confirm was called when an update task is in progress. This is not allowed, wait for updates to complete before calling confirm."
+                let error = PaymentSheetError.embeddedPaymentElementConfirmFailed(message: errorMessage)
```

Comment:
can we use integrationError instead here?

Reply:
Yeah guess we can, we should align on how to do this in the future given the recent feedback here of preferring specific errors: https://docs.google.com/document/d/1Iprd_nypdC5XSnL8XWwn1DmhgXKib3F4eLiuv4Nlr7g/edit?tab=t.26tjl61qwqtm#bookmark=id.5wogq6eqo5tm